hack/generator/pkg/astmodel/storage_package_reference.go

```diff
@@ -29,8 +32,16 @@ func MakeStoragePackageReference(local LocalPackageReference) StoragePackageRefe
 }
 
 // String returns the string representation of the package reference
-func (spr StoragePackageReference) String() string {
-	return fmt.Sprintf("storage:%v", spr.PackagePath())
+func (s StoragePackageReference) String() string {
+	return fmt.Sprintf("storage:%v", s.PackagePath())
+}
+
+// IsPreview returns true if this package reference is a preview
+func (s StoragePackageReference) IsPreview() bool {
+	lc := strings.ToLower(s.version)
+	return strings.Contains(lc, "alpha") ||
```

Comment:
Is there a way we could have this logic shared among the different package references? Possibly by defining a module-level function that does the check and then having each interface implementation call it? For example I see that you define `specialVersionIdentifiers` but then don't use it here -- a shared function could make use of that collection possibly and save a good bit of duplication?

Plus then there's only 1 place to change if we need to add/remove strings to it.

Reply:
Good idea, done.